src/js/store/factory/FacetLayout.ts

```diff
@@ -0,0 +1,41 @@
+import {Map, Record, RecordOf} from 'immutable';
+
+/**
+ * Layouts align multiple groups
+ */
+export interface FacetLayout {
```

Comment:
is there enough overlap between FacetLayout and Layout that we can think about e.g.:
- a common interface that both of them "inherit" from via the typescript `&` (intersection type) operator
- a "parent type" defined as the `|` (union type) of the two?

i'm not suggesting either particular choice is more right here but consider whether or not that might simplify things in some places (it also might not simplify things at all, in which case feel free to reject this idea)

Reply:
Interesting, they are similar in concept but very different in implementation and how they are used in Lyra and the ultimate vega spec.  Perhaps worth discussing further